docs/_static/link_gen/link.js

```diff
@@ -117,6 +117,18 @@ function changeTab(div) {
     }
 }
 
+/**
+ * Return name of directory git will clone given repo to.
+ *
+ * nbgitpuller needs to redirect users to *inside* the directory it
+ * just cloned. We copy the logic git itself uses to determine that.
+ * See https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L276
+ */
+function generateCloneDirectoryName(gitCloneUrl) {
+    var lastPart = gitCloneUrl.split('/').slice(-1)[0];
+    return lastPart.split(':').slice(-1)[0].replace(/(\.git|\.bundle)?/, '');
```

Comment:
Splitting at `:` is for when the branch name is passed with the repo url?
There is a `branch` entry in the link generator page, so passing the branch with the repo url won't happen too often *I think*.

Reply:
If you have a clone URL like `git@github.com:my-repo`, git will put it in `my-repo` directory. There are no `/` here! I got this from [this comment](https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L271) in the git repository

Reply:
It makes sense now. Thanks!